ddl/ddl_api.go

```diff
@@ -1814,6 +1814,13 @@ func (d *ddl) CreateTable(ctx sessionctx.Context, s *ast.CreateTableStmt) (err e
 	if err != nil {
 		return errors.Trace(err)
 	}
+	if tbInfo.TempTableType != model.TempTableNone {
+		for _, column := range tbInfo.Columns {
+			if column.IsGenerated() && !column.GeneratedStored {
+				return ErrOptOnTemporaryTable.GenWithStackByArgs("virtual columns")
```

Comment:
Is it really unable to support virtual columns? Would you have a try?

Reply:
If it's able to support it, we don't need to forbid it.

Reply:
i test it on my compute, if the column is virtual, it match the condition in line 1819. why don't need to forbid it?